src/datachain/query/dataset.py

```diff
@@ -1097,26 +1091,43 @@ def __init__(
         self.temp_table_names: list[str] = []
         self.dependencies: set[DatasetDependencyType] = set()
         self.table = self.get_table()
-        self.starting_step: StartingStep
+        self.query_step: Optional[QueryStep] = None
```

Comment:
I am wonder if we can think of better name for `query_step` attribute. I have a lot of questions lower in the code and keep confusing about this name. Should we, may be, keep `starting_step` name as it is more verbose? 🤔

Reply:
Thank you! I really think it will be more verbose 🙏